### PR TITLE
Skip transition of unverified -> verified when email is disabled (new "proven" event)

### PR DESCRIPTION
**Pull request by jorissteyn**
Institutions which do not require email verification found 'Email
verified' events in the audit log.

Previous implementation was internally creating an unverified second
factor and using the verify-email event to transition it to verified,
causing this unwanted effect. The new implementation skips the
creationg of unverified second factors altogether.

So, with email verification:

    user proves possession -> create unverified -> user verifies email -> transition to verified

Previously, without email verification

    user proves possession -> create unverified -> transition to verified

New implementation, without email verification

    user proves possession -> create verified

This is achieved by creating new 'proven and verified' events for
tokens which are implicitly verified by the Idp. These events are
handled differently from the regular 'proven' events, and to maintain
compatibility with the previous way of working, handling of those
regular events is unchanged.

### CODE REVIEW

File: src/Surfnet/Stepup/Identity/Event/GssfPossessionProvenAndVerifiedEvent.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
```
Comment (MKodde) [inline src/Surfnet/Stepup/Identity/Event/GssfPossessionProvenAndVerifiedEvent.php]: Nitpick: ~~2014~~ 2018

Comment (MKodde) [reply src/Surfnet/Stepup/Identity/Event/GssfPossessionProvenAndVerifiedEvent.php]: (Applies to other files)

File: src/Surfnet/Stepup/Identity/Event/YubikeyPossessionProvenAndVerifiedEvent.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+     * @param string                  $registrationCode
+     */
+    public function __construct(
+        IdentityId $identityId,
+        Institution $institution,
+        SecondFactorId $secondFactorId,
+        YubikeyPublicId $yubikeyPublicId,
+        CommonName $commonName,
+        Email $email,
+        DateTime $registrationRequestedAt,
```
Comment (MKodde) [inline src/Surfnet/Stepup/Identity/Event/YubikeyPossessionProvenAndVerifiedEvent.php]: `registrationRequestedAt` and `registrationCode` are not yet used? Probably due to the WIP status..

File: src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/AuditLogRepository.php [hunk 3]
```diff
@@ -36,7 +36,9 @@ class AuditLogRepository extends EntityRepository
         'Surfnet\Stepup\Identity\Event\YubikeySecondFactorBootstrappedEvent',
         'Surfnet\Stepup\Identity\Event\GssfPossessionProvenEvent',
         'Surfnet\Stepup\Identity\Event\PhonePossessionProvenEvent',
-        'Surfnet\Stepup\Identity\Event\YubikeyPossessionProvenEvent',
+        'Surfnet\Stepup\Identity\Event\YubikeyPossessionProvenAndVerifiedEvent',
```
Comment (MKodde) [inline src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/AuditLogRepository.php]: Do you know why u2f events are not logged?

File: src/Surfnet/Stepup/Identity/Event/YubikeyPossessionProvenAndVerifiedEvent.php (hunk 2 above)
Comment (MKodde) [reply src/Surfnet/Stepup/Identity/Event/YubikeyPossessionProvenAndVerifiedEvent.php]: This applies to other events

File: src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/AuditLogRepository.php (hunk 3 above)
Comment (MKodde) [reply src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/AuditLogRepository.php]: Discussed IRL, can be added in a new story if need be.

File: src/Surfnet/Stepup/Identity/Event/GssfPossessionProvenAndVerifiedEvent.php (hunk 1 above)
Comment (jorissteyn) [reply src/Surfnet/Stepup/Identity/Event/GssfPossessionProvenAndVerifiedEvent.php]: Fixed!

File: src/Surfnet/Stepup/Identity/Event/YubikeyPossessionProvenAndVerifiedEvent.php (hunk 2 above)
Comment (jorissteyn) [reply src/Surfnet/Stepup/Identity/Event/YubikeyPossessionProvenAndVerifiedEvent.php]: I've checked and fixed all constructor assignments.

File: src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/AuditLogRepository.php (hunk 3 above)
Comment (jorissteyn) [reply src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/AuditLogRepository.php]: Checking with Pieter, I'll create a new story if it's unintentional.

**Pull request merged**